common/src/main/java/com/revenuecat/purchases/common/networking/ETagManager.kt

```diff
@@ -0,0 +1,112 @@
+package com.revenuecat.purchases.common.networking
+
+import android.content.Context
+import android.content.SharedPreferences
+import com.revenuecat.purchases.common.sha1
+import org.json.JSONObject
+
+private const val NOT_MODIFIED_RESPONSE_CODE = 304
+private const val INTERNAL_SERVER_ERROR_RESPONSE_CODE = 500
+private typealias RequestHash = String
+
+data class HTTPResultWithETag(
+    val eTag: String,
+    val httpResult: HTTPResult
+) {
+    fun serialize(): String {
+        val jsonObject = JSONObject()
+        jsonObject.put("eTag", eTag)
+        jsonObject.put("httpResult", httpResult.serialize())
+        return jsonObject.toString()
+    }
+
+    companion object {
+        fun deserialize(serialized: String): HTTPResultWithETag {
+            val jsonObject = JSONObject(serialized)
+            val eTag = jsonObject.getString("eTag")
+            val serializedHTTPResult = jsonObject.getString("httpResult")
+            return HTTPResultWithETag(eTag, HTTPResult.deserialize(serializedHTTPResult))
+        }
+    }
+}
+
+class ETagManager(
+    private val prefs: SharedPreferences
+) {
+    private var hashes: Map<HTTPRequest, RequestHash> = emptyMap()
```

Comment:
I'd name this `hashesByHttpRequest` to make the storing and keys very clear at first sight